foodhub/accounts/views.py

```diff
@@ -0,0 +1,239 @@
+        return Response(serializer.data)
+
+    @action(detail=True)
+    def orders(self, request, pk=None):
+        merchant = Merchant.objects.filter(pk=pk).first()
+        orders = Order.objects.filter(merchant=merchant)
+        serializer = OrderSerializer(orders, many=True)
+        return Response(serializer.data)
+
+
+class ItemViewSet(viewsets.ModelViewSet):
+    """
+    API view to perform CRUD operations on Item
+    """
+    serializer_class = ItemSerializer
+    queryset = Item.objects.select_related('merchant').all()
+    permission_classes = [IsAuthenticated]
+
+    def create(self, request, *args, **kwargs):
+        response = super().create(request, *args, **kwargs)
+        log.msg('Create Item Request', req=request.data)
+        return response
+    def update(self, request, *args, **kwargs):
+        response = super().create(request, *args, **kwargs)
+        log.msg('Update Item Request', req=request.data)
+        return response
+
+
+class StoreViewSet(viewsets.ModelViewSet):
+    """
+    API view to perform CRUD operations on Store
+    """
+    serializer_class = StoreSerializer
+    queryset = Store.objects.select_related(
+        'merchant').prefetch_related('items')
+    permission_classes = [IsAuthenticated]
+
+
+    def create(self, request, *args, **kwargs):
+        response = super().create(request, *args, **kwargs)
+        log.msg('Create Store Request', req=request.data)
+        return response
+    def update(self, request, *args, **kwargs):
+        response = super().create(request, *args, **kwargs)
+        log.msg('Update Store Request', req=request.data)
+        return response
+
+    @action(detail=True)
+    def orders(self, request, pk=None):
+        store = Store.objects.filter(pk=pk).first()
+        orders = Order.objects.filter(store=store)
+        serializer = OrderSerializer(orders, many=True)
+        return Response(serializer.data)
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    """
+    API view to perform CRUD operations on Order
+    """
+    serializer_class = OrderSerializer
+    queryset = Order.objects.select_related(
+        'merchant', 'store').prefetch_related('items')
+    permission_classes = [IsAuthenticated]
+    pagination_class = api_settings.DEFAULT_PAGINATION_CLASS
+    def create(self, request, *args, **kwargs):
+        response = super().create(request, *args, **kwargs)
+        log.msg('Create Order Request', req=request.data)
+        return response
+    def update(self, request, *args, **kwargs):
+        response = super().create(request, *args, **kwargs)
+        log.msg('Update Order Request', req=request.data)
+        return response
+
+    @property
+    def paginator(self):
+        """
+        The paginator instance associated with the view, or `None`.
+        """
+        user_agent = self.request.headers.get('User-Agent', None)
+        log.info('INSIDE paginator',user_agent = user_agent)
```

Comment:
just a sugg, try to add events as snake_case.

Reply:
As per requirement i have updated my views.py file